hadoop-hdds/rocks-native/src/test/java/org/apache/hadoop/hdds/utils/db/managed/TestManagedSSTDumpIterator.java

```diff
@@ -0,0 +1,109 @@
+package org.apache.hadoop.hdds.utils.db.managed;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+/**
+ * Test for ManagedSSTDumpIterator.
+ */
+public class TestManagedSSTDumpIterator {
+
+  private void testSSTDumpIteratorWithKeys(
+      Map<Pair<String, Integer>, String> records)
+      throws Exception {
+    Map<Pair<String, Integer>, String> keys = records instanceof TreeMap ?
```

Comment:
1. These tests could be changed in to parameterized test.
2. If you expect `records` to be an instance of `TreeMap`, wouldn't it be better for requested to send a `TreeMap` itself.
3. I don't get (or like) idea of using `Pair<String, Integer>` as key. It could be simplified if you use parameterized tests.

Reply:
Just to clarify, I meant parameterized test of Junit-5 not Junit-4.